cmd/entrypoint/main.go

```diff
@@ -55,15 +71,20 @@ type RealWaiter struct{ waitFile string }
 
 var _ entrypoint.Waiter = (*RealWaiter)(nil)
 
-func (*RealWaiter) Wait(file string) {
+func (*RealWaiter) Wait(file string) error {
 	if file == "" {
-		return
+		return nil
 	}
 	for ; ; time.Sleep(time.Second) {
+		// Watch for the post file
 		if _, err := os.Stat(file); err == nil {
-			return
+			return nil
 		} else if !os.IsNotExist(err) {
-			log.Fatalf("Waiting for %q: %v", file, err)
+			return fmt.Errorf("Waiting for %q: %v", file, err)
```

Comment:
is this now going to return an error right away and make it not wait?

Reply:
It doesn't change the behavior as `log.Fatalf` would have called `os.Exit` (and killed the process). We just "control" where the `kill` happens :sweat_smile: